app/src/main/java/org/jellyfin/androidtv/ui/search/SearchRepository.kt

```diff
@@ -0,0 +1,46 @@
+package org.jellyfin.androidtv.ui.search
+
+import org.jellyfin.sdk.api.client.ApiClient
+import org.jellyfin.sdk.api.client.exception.ApiClientException
+import org.jellyfin.sdk.api.client.extensions.itemsApi
+import org.jellyfin.sdk.model.api.BaseItemDto
+import org.jellyfin.sdk.model.api.BaseItemKind
+import org.jellyfin.sdk.model.api.ItemFields
+import java.io.IOException
+
+interface SearchRepository {
+    suspend fun search(searchTerm: String, itemTypes: Collection<BaseItemKind>): List<BaseItemDto>
+}
+
+class SearchRepositoryImpl(
+        private val apiClient: ApiClient
+) : SearchRepository {
+    companion object {
+        private const val QUERY_LIMIT = 25
+    }
+
+    override suspend fun search(
+            searchTerm: String,
+            itemTypes: Collection<BaseItemKind>
+    ): List<BaseItemDto> {
```

Comment:
If we use Result<> here instead of emptyList on failure we can deal with errors in the usage side. Right now errors are ignored (not even logged). The interface would use `Result<List<BaseItemDto>>` as return type and this function would use either 
`Result.success(result.content.items.orEmpty())` or `Result.failure(e)` (in the catch)

Reply:
I'll look into this, at first glance I didn't see a way to do that with `org.jellyfin.sdk.api.client.Response`